include/libcyphal/transport/scattered_buffer.hpp

```diff
@@ -15,53 +15,99 @@ namespace libcyphal
 namespace transport
 {
 
+/// @brief Represents a buffer that could be scattered across multiple memory regions of an abstract storage.
+///
 /// The buffer is movable but not copyable because copying the contents of a buffer is considered wasteful.
 /// The buffer behaves as if it's empty if the underlying implementation is moved away.
 ///
 class ScatteredBuffer final
 {
     // 91C1B109-F90E-45BE-95CF-6ED02AC3FFAA
-    using InterfaceTypeIdType = cetl::
+    using IStorageTypeIdType = cetl::
         type_id_type<0x91, 0xC1, 0xB1, 0x09, 0xF9, 0x0E, 0x45, 0xBE, 0x95, 0xCF, 0x6E, 0xD0, 0x2A, 0xC3, 0xFF, 0xAA>;
 
 public:
-    static constexpr std::size_t ImplementationFootprint = sizeof(void*) * 8;
+    /// @brief Defines maximum size (aka footprint) of the storage variant.
+    ///
+    static constexpr std::size_t StorageVariantFootprint = sizeof(void*) * 8;
 
-    class Interface : public cetl::rtti_helper<InterfaceTypeIdType>
+    /// @brief Defines storage interface for the scattered buffer.
+    ///
+    /// @see ScatteredBuffer::ScatteredBuffer(AnyStorage&& any_storage)
+    ///
+    class IStorage : public cetl::rtti_helper<IStorageTypeIdType>
```

Comment:
Here and there (including in the design doc) we use word "storage" when we talk about Lizard-specific implementation of actual (potentially) scattered storage this "facade" `ScatteredBuffer` is using. So, I decided to rename nested `Interface` to IMO more meaningful `IStorage`.